curses.h

```diff
@@ -34,11 +34,16 @@ PDCurses portable platform definitions list:
 #define XOPEN           1      /* X/Open Curses routines */
 #define SYSVcurses      1      /* System V Curses routines */
 #define BSDcurses       1      /* BSD Curses routines */
-#define CHTYPE_LONG     1      /* size of chtype; long */
+#if defined( CHTYPE_32)
+   #define CHTYPE_LONG     1      /* chtypes will be 32 bits */
+#elif !defined( CHTYPE_16)
+   #define CHTYPE_LONG     2      /* chtypes will be (default) 64 bits */
```

Comment:
Hi Luke,

On 2016-03-23 00:26, Luke-Jr wrote:
> In curses.h <https://github.com/wmcbrine/PDCurses/pull/6#discussion_r57108353>:
>
>> @@ -34,11 +34,16 @@ PDCurses portable platform definitions list:
>>  #define XOPEN           1      /* X/Open Curses routines */
>>  #define SYSVcurses      1      /* System V Curses routines */
>>  #define BSDcurses       1      /* BSD Curses routines */
>> -#define CHTYPE_LONG     1      /* size of chtype; long */
>> +#if defined( CHTYPE_32)
>> +   #define CHTYPE_LONG     1      /* chtypes will be 32 bits */
>> +#elif !defined( CHTYPE_16)
>> +   #define CHTYPE_LONG     2      /* chtypes will be (default) 64 bits */
>
> What if CHTYPE_16 is defined?

    In that case,  CHTYPE_LONG is left undefined.

    There's some discussion of this in curses.h (search for "Originally,
PDCurses used a short" and read from there.)  Until Win32a came along,
your options were to leave CHTYPE_LONG undefined,  in which case you
got 16-bit chtypes and some limited attributes and colors;  or you
could #define CHTYPE_LONG 1 and get 32-bit chtypes,  basic Unicode
support (i.e.,  up to U+FFFF),  and plenty of colors and attributes.

    Since this was a binary decision,  it made sense to have the options
be "either CHTYPE_LONG is defined,  or it isn't."  Making it a ternary
choice,  with 64-bit chtypes as an option,  threw a spanner into the
works.  For backward compatibility,  I just set things up so that
option was enabled with #define CHTYPE_LONG 2.

-- Bill

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/wmcbrine/PDCurses/pull/6/files/c1ab1e40dc35a905edd65e5263ce9d7029d5e452#r57108353>
>